engines/e_capi.c

```diff
@@ -472,10 +476,9 @@ static int capi_init(ENGINE *e)
     }
 # endif
 
-    /* See if we support AES CSP */
-
-    if (CryptAcquireContext(&hprov, NULL, NULL, PROV_RSA_AES,
-                            CRYPT_VERIFYCONTEXT)) {
+    /* See if there is RSA+AES CSP */
+    if (CryptAcquireContextW(&hprov, NULL, rsa_aes_cspname, PROV_RSA_AES,
```

Comment:
> what if the default if 3rd argument remains NULL?

Who knows, it might find some *other* PROV_RSA_AES-capable CSP. To recap. `RPOV_*` is kind of "capability" and there can be multiple CSPs with same capability. At the same time one has to keep in mind that private key can be so to say bound to CSP, i.e. if it's served by CSP X with "capability" N, it's not given that you can replace X with Y. That's the whole point with #2920. And whole idea behind this suggestion is to play PROV_RSA_FULL->PROV_RSA_AES trick only with CSPs known to be interchangeable.